Dockerfile

```diff
@@ -1,17 +1,56 @@
-FROM ubuntu:xenial
+# syntax=docker/dockerfile:experimental
 
-RUN apt-get update && apt-get install --yes nginx
+# Build stage: Install ruby dependencies
+# ===
+FROM ruby:2.5 AS build-site
+WORKDIR /srv
+ADD . .
+RUN bundle install
+RUN bundle exec jekyll build
+
+# Build stage: Install yarn dependencies
+# ===
+FROM node:12-slim AS yarn-dependencies
+WORKDIR /srv
+ADD package.json .
+RUN --mount=type=cache,target=/usr/local/share/.cache/yarn yarn install
 
-# Set git commit ID
-ARG REVISION_ID
-RUN test -n "${REVISION_ID}"
+# Build stage: Run "yarn run build-js"
+# ===
+FROM yarn-dependencies AS build-js
+WORKDIR /srv
+COPY . .
+RUN yarn run build-js
 
-# Copy over files
+# Build stage: Run "yarn run build-css"
+# ===
+FROM yarn-dependencies AS build-css
 WORKDIR /srv
-ADD _site .
+COPY . .
+RUN yarn run build-css
+
+# Build the production image
+# ===
+FROM ubuntu:focal
+
+# Set up environment
+ENV LANG C.UTF-8
+WORKDIR /srv
+
+# Install nginx
+RUN apt-get update && apt-get install --no-install-recommends --yes nginx
+
+# Import code, build assets and mirror list
+RUN rm -rf package.json yarn.lock .babelrc webpack.config.js Gemfile Gemfile.lock nginx.conf requirements.txt
```

Comment:
The `requirements.txt` should be in the `.dockerignore` file since it's only used in the circle CI job.